utility_analysis_new/dpengine.py

```diff
@@ -0,0 +1,145 @@
+from pipeline_dp import DataExtractors
+from pipeline_dp import NoiseKind
+from pipeline_dp import contribution_bounders
+from pipeline_dp.contribution_bounders import ContributionBounder
+from dataclasses import dataclass
+
+
+class UtilityAnalysisDPEngine(DPEngine):
+
+    def __init__(self, budget_accountant: 'BudgetAccountant',
+                 backend: 'PipelineBackend'):
+        super().__init__(budget_accountant, backend)
+
+    def aggregate(self,
+                  col,
+                  params: pipeline_dp.AggregateParams,
+                  data_extractors: DataExtractors,
+                  public_partitions=None):
+        return super().aggregate(col, params, data_extractors,
+                                 public_partitions)
+
+    def _aggregate(self, col, params: pipeline_dp.AggregateParams,
+                   data_extractors: DataExtractors, public_partitions):
+        assert params.custom_combiners is None, "Cumtom combiners are not supported"
+        assert params.metrics == [
+            pipeline_dp.Metrics.COUNT
+        ], f"Supported only count metrics, metrics={params.metrics}"
+        assert public_partitions is not None, "Only public partitions supported"
+        assert params.contribution_bounds_already_enforced is False, "Utility Analysis when contribution bounds are already enforced is not supported"
+
+        # TODO: Implement custom combiner
+        combiner = self._create_compound_combiner(params)
+        # Extract the columns.
+        col = self._backend.map(
+            col, lambda row: (data_extractors.privacy_id_extractor(row),
+                              data_extractors.partition_extractor(row),
+                              data_extractors.value_extractor(row)),
+            "Extract (privacy_id, partition_key, value))")
+        # col : (privacy_id, partition_key, value)
+        contribution_bounder = self._create_contribution_bounder(params)
+        col = contribution_bounder.bound_contributions(
+            col, params, self._backend, self._current_report_generator,
+            combiner.create_accumulator)
+        # col : ((privacy_id, partition_key), accumulator)
+
+        col = self._backend.map_tuple(col, lambda pid_pk, v: (pid_pk[1], v),
+                                      "Drop privacy id")
+        # col : (partition_key, accumulator)
+
+        # Compute DP metrics.
+        self._add_report_stages(combiner.explain_computation())
+        col = self._backend.map_values(col, combiner.compute_metrics,
+                                       "Compute DP` metrics")
+
+        return col
+
+    def _create_contribution_bounder(
+        self, params: pipeline_dp.AggregateParams
+    ) -> contribution_bounders.ContributionBounder:
+        """Creates Utility Analysis ContributionBounder."""
+        return UtilityAnalysisSamplingCrossAndPerPartitionContributionBounder()
+
+
+class UtilityAnalysisSamplingCrossAndPerPartitionContributionBounder(
```

Comment:
1. Let's move it utility_analysis_new/contribution_bounders.py
2. `UtilityAnalysisSamplingCrossAndPerPartitionContributionBounder` feels too long, it seems better to drop UtiliatyAnalysis from the name. All classes are referenced with module name, so we can write like here
`import utility_analysis.contribution_bounders as utility_contribution_bounders

bounder = utility_contribution_bounders.SamplingCrossAndPerPartitionContributionBounder()
`